homeassistant/helpers/service.py

```diff
@@ -360,3 +363,47 @@ def async_register_admin_service(
     hass.services.async_register(
         domain, service, admin_handler, schema
     )
+
+
+@bind_hass
+@ha.callback
+def verify_domain_control(hass: HomeAssistantType, domain: str) -> Callable:
+    """Ensure permission to access any entity under domain in service call."""
+    def decorator(service_handler: Callable) -> Callable:
+        """Decorate."""
+        if not asyncio.iscoroutinefunction(service_handler):
+            raise HomeAssistantError(
+                'Cannot decorate service handler: {0}'.format(service_handler))
```

Comment:
No need to add name, this exception is during startup and will have correct line number. Instead, include why you cannot decorate to educate users.

```suggestion
                'Can only decorate async functions.')
```